Tests/Command/CloneSiteCommandTest.php

```diff
@@ -0,0 +1,199 @@
+        $newPage2 = $page2;
+        $newPage2->setTitle('Copy of Page 2')->shouldBeCalled();
+        $newPage2->setSite($destSite)->shouldBeCalled();
+        $newPage2->setParent($newPage1)->shouldBeCalled();
+
+        $newPage1->setTarget($newPage2)->shouldBeCalled();
+
+        $this->pageManager->save($newPage2)->shouldBeCalled();
+        $this->pageManager->save($newPage2, true)->shouldBeCalled();
+
+        $block = $this->prophesize('Sonata\PageBundle\Model\PageBlockInterface');
+        $block->getId()->willReturn(4711);
+        $block->getParent()->willReturn(null);
+
+        $this->blockManager->findBy(array(
+           'page' => $page1,
+        ))->willReturn(array());
+
+        $this->blockManager->findBy(array(
+           'page' => $page2,
+        ))->willReturn(array(
+            $block->reveal(),
+        ));
+
+        // Replace this with new mock, when cloing is supported in prophecies
+        $newBlock = $block;
+        $block->setPage($newPage2)->shouldBeCalled();
+
+        $this->blockManager->save($newBlock)->shouldBeCalled();
+
+        $command = $this->application->find('sonata:page:clone-site');
+        $commandTester = new CommandTester($command);
+        $commandTester->execute(array(
+            'command' => $command->getName(),
+            '--source-id' => 23,
+            '--dest-id' => 42,
+            '--prefix' => 'Copy of ',
+            '--only-hybrid' => true,
+        ));
+
+        $this->assertRegExp('@done!@', $commandTester->getDisplay());
+    }
+
+    public function testExecuteNoSourceId()
+    {
+        $this->setExpectedException('\InvalidArgumentException', 'Please provide a "--source-id=SITE_ID" option.');
+
+        $this->siteManager->findAll()->willReturn(array());
+
+        $command = $this->application->find('sonata:page:clone-site');
+        $commandTester = new CommandTester($command);
+        $commandTester->execute(array(
+            'command' => $command->getName(),
+            '--dest-id' => 42,
+            '--prefix' => 'Copy of ',
+        ));
+
+        $this->assertRegExp('@Writing cache file ...\s+done!@', $commandTester->getDisplay());
+    }
+
+    public function testExecuteNoDestId()
+    {
+        $this->setExpectedException('\InvalidArgumentException', 'Please provide a "--dest-id=SITE_ID" option.');
+
+        $this->siteManager->findAll()->willReturn(array());
+
+        $command = $this->application->find('sonata:page:clone-site');
+        $commandTester = new CommandTester($command);
+        $commandTester->execute(array(
+            'command' => $command->getName(),
+            '--source-id' => 23,
+            '--prefix' => 'Copy of ',
+        ));
+
+        $this->assertRegExp('@Writing cache file ...\s+done!@', $commandTester->getDisplay());
+    }
+
+    public function testExecuteNoPrefix()
+    {
+        $this->setExpectedException('\InvalidArgumentException', 'Please provide a "--prefix=PREFIX" option.');
```

Comment:
using this is deprecated IMO, shouldn't we use annotations?

same above? @greg0ire @core23